BuddySave/CloudManager.cs

```diff
@@ -0,0 +1,59 @@
+namespace BuddySave;
+
+public class CloudManager : ICloudManager
+{
+    public void UploadSave(GameSave gameSave)
+    {
+        CopyOverSaves(gameSave.LocalPath, gameSave.CloudPath);
+    }
+
+    public void DownloadSave(GameSave gameSave)
+    {
+        CopyOverSaves(gameSave.CloudPath, gameSave.LocalPath);
+    }
+
+    public bool LockExists(GameSave save)
+    {
+        var lockPath = GetLockPath(save);
+        return File.Exists(lockPath);
+    }
+
+    public async Task CreateLock(GameSave save)
+    {
+        var lockPath = GetLockPath(save);
+        await using var fileWriter = new FileStream(lockPath, FileMode.CreateNew);
+        await using var streamWriter = new StreamWriter(fileWriter);
+        await streamWriter.WriteLineAsync(DateTime.Now.ToString("o"));
+    }
+
+    public void DeleteLock(GameSave save)
+    {
+        var lockPath = GetLockPath(save);
+        File.Delete(lockPath);
+    }
+
+    private static void CopyOverSaves(string sourcePath, string destinationPath)
+    {
```

Comment:
```suggestion
    {
        if(!Directory.Exists(sourcePath))
        {
            // notify client that source does not exist or exception
            return;
        }
```

Reply:
I've decided to throw exceptions instead - we don't want it to run, if config is wrong